lib/Controller/SignalingController.php

```diff
@@ -104,6 +109,54 @@ public function getSettings(): DataResponse {
 		return new DataResponse($this->config->getSettings($this->userId));
 	}
 
+	/**
+	 * Only available for logged in users because guests can not use the apps
+	 * right now.
+	 *
+	 * @param int $serverId
+	 * @return DataResponse
+	 */
+	public function getWelcomeMessage(int $serverId): DataResponse {
+		$signalingServers = $this->config->getSignalingServers();
+		if (empty($signalingServers) || !isset($signalingServers[$serverId])) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+
+		$url = rtrim($signalingServers[$serverId]['server'], '/');
+
+		if (strpos($url, 'wss://') === 0) {
+			$url = 'https://' . substr($url, 6);
+		}
+
+		if (strpos($url, 'ws://') === 0) {
+			$url = 'http://' . substr($url, 5);
+		}
+
+		$client = $this->clientService->newClient();
+		try {
+			$response = $client->get($url . '/api/v1/welcome', [
+				'verify' => (bool) $signalingServers[$serverId]['verify'],
+			]);
+
+			$body = $response->getBody();
+
+			$data = json_decode($body, true);
+			if (!is_array($data)) {
+				return new DataResponse(['error' => 'JSON_INVALID'], Http::STATUS_INTERNAL_SERVER_ERROR);
+			}
+
+			if (!isset($data['version'])) {
+				return new DataResponse(['error' => 'VERSION_TOO_OLD'], Http::STATUS_INTERNAL_SERVER_ERROR);
```

Comment:
The version is provided in the welcome message even in the 0.0.3 version of the external signaling server, so this error code is probably unneeded in the real world (although, in that case, it would be better to add again the check removed in fe4ace7 even if the `VERSION_TOO_OLD` error is no longer sent).

Reply:
I didn't see it in 0.0.9, but maybe I checked the wrong message then.
But well, let's just keep it instead of breaking